t/plugin/openid-connect6.t

```diff
@@ -155,3 +155,213 @@ passed
     }
 --- response_body
 passed
+
+
+
+=== TEST 4: Update route with Keycloak introspection endpoint and introspection addon headers.
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "openid-connect": {
+                                "client_id": "course_management",
+                                "client_secret": "d1ec69e9-55d2-4109-a3ea-befa071579d5",
+                                "discovery": "http://127.0.0.1:8080/realms/University/.well-known/openid-configuration",
+                                "redirect_uri": "http://localhost:3000",
+                                "ssl_verify": false,
+                                "timeout": 10,
+                                "bearer_only": true,
+                                "realm": "University",
+                                "introspection_endpoint_auth_method": "client_secret_post",
+                                "introspection_endpoint": "http://127.0.0.1:8080/realms/University/protocol/openid-connect/token/introspect",
+                                "introspection_addon_headers": {
+                                    "X-Addon-Header-A": "VALUE",
```

Comment:
hardcoding values in configuration doesn't make sense, I think the configuration should just specify header fields which would be extracted from the original request.

Reply:
do you mean change the headers from the original request?

Reply:
no, the plugin configuration should only specify which headers should be sent. APISIX should extract those headers from the request and then add them to the introspection request.

Reply:
I thinking it's hard to decide which headers should be sent, maybe we should just keep it open here and leave it to user customization?